tools/teamcity-diff-check/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+)
+
+var serviceFile = flag.String("service_file", "services_ga", "kotlin service file to be parsed")
+
+func serviceDifference(gS, tS []string) []string {
+	t := make(map[string]struct{}, len(tS))
+	for _, s := range tS {
+		t[s] = struct{}{}
+	}
+
+	var diff []string
+	for _, s := range gS {
+		if _, found := t[s]; !found {
+			diff = append(diff, s)
+		}
+	}
+
+	return diff
+}
+
+func main() {
+	flag.Parse()
+
+	file, err := os.Open(*serviceFile + ".txt")
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	defer file.Close()
+
+	googleServices := []string{}
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		googleServices = append(googleServices, scanner.Text())
+	}
+
+	////////////////////////////////////////////////////////////////////////////////
+
+	f, err := os.Open(fmt.Sprintf("../../mmv1/third_party/terraform/.teamcity/components/inputs/%s", *serviceFile+".kt"))
```

Comment:
This would be good to follow, this would also require another flag to include the use of the `service_file.txt` which is where all services are stored from the `git diff` command. This is due to the `service_file` flag also being used for the `.txt` file also

They could be separated to be two different flags: `-servicekt_path` and `-service_txt`

For now we can save this for another time 👍🏼